File: VB/VB/Form1.vb
Imports Microsoft.VisualBasic
Imports System
Imports System.Collections.Generic
Imports System.ComponentModel
Imports System.Data
Imports System.Drawing
Imports System.Text
Imports System.Windows.Forms
Imports DevExpress.XtraGrid.Views.Grid
Imports DevExpress.XtraGrid

Namespace WindowsApplication1
	Partial Public Class Form1
		Inherits Form

		Public Sub New()
			InitializeComponent()
			gridControl1.DataSource = DataGenerator.GetData()
			AddHandler gridControl1.ViewRegistered, AddressOf gridControl1_ViewRegistered
			AddHandler gridView1.MasterRowEmpty, AddressOf gridView1_MasterRowEmpty
			gridView1.OptionsDetail.AllowExpandEmptyDetails = True
		End Sub

		Private Sub gridView1_MasterRowEmpty(ByVal sender As Object, ByVal e As MasterRowEmptyEventArgs)
			e.IsEmpty = False
		End Sub

		Private Sub gridControl1_ViewRegistered(ByVal sender As Object, ByVal e As ViewOperationEventArgs)
			TryCast(e.View, GridView).OptionsView.NewItemRowPosition = NewItemRowPosition.Bottom
		End Sub
	End Class
End Namespace

File: CS/Form1.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Text;
using System.Windows.Forms;
using DevExpress.XtraGrid.Views.Grid;
using DevExpress.XtraGrid;

namespace WindowsApplication1
{
    public partial class Form1 : Form
    {

        public Form1()
        {
            InitializeComponent();
            gridControl1.DataSource = DataGenerator.GetData();
            gridControl1.ViewRegistered += gridControl1_ViewRegistered;
            gridView1.MasterRowEmpty += gridView1_MasterRowEmpty;
            gridView1.OptionsDetail.AllowExpandEmptyDetails = true;
        }

        void gridView1_MasterRowEmpty(object sender, MasterRowEmptyEventArgs e)
        {
            e.IsEmpty = false;
        }

        void gridControl1_ViewRegistered(object sender, ViewOperationEventArgs e)
        {
            (e.View as GridView).OptionsView.NewItemRowPosition = NewItemRowPosition.Bottom;
        }
    }
}